src/ruby/puppetserver-lib/puppet/server/http_response.rb

```diff
@@ -0,0 +1,48 @@
+require 'puppet/http/response'
+
+class Puppet::Server::HttpResponse < Puppet::HTTP::Response
+  attr_reader :url
+
+  def initialize(java_response, url)
+    @java_response = java_response
+    @url = url
```

Comment:
Looks like this should also be a URI object that includes the encoded query params, correct?

Reply:
Yep. Since https://github.com/puppetlabs/puppetserver/pull/2419/files#diff-5805ba6eb4857e236651f3c3fcdffeb0R72 passes in a URL encoded string, I think you could do `@url = URI(url)` here? Or maybe clearer to have the caller do that and pass the `URI` object.

Reply:
I'd rather have it be called "URL" because we really only expect this to ever be an HTTP url, since this is explicitly an HTTP response class. But I'll add a doc comment.